### DISCUSSION TITLE
ParseExpressionPipe update error since 6.98

### DISCUSSION THREAD

**Discussion by Benjamin8585** [TypedProject/ts-express-decorators#2035] [Q&A]
Hello,
Since this version ParseExpressionPipe has been updated:

```typescript
  protected getKey(param: JsonParameterStore) {
    let {expression, paramType, dataPath} = param;
    paramType = paramType || param.parameter.get("in").toUpperCase();

    if (expression && paramType === ParamTypes.HEADER) {
      expression = String(expression).toLowerCase();
    }

    return [dataPath, expression].filter(Boolean).join(".");
  }
```

My server crashes because ```param.parameter.get("in")``` is undefined (custom ContextPipe)
What does this "in" property refers to?

Thanks

**Comment by Romakita** [comment]
Please update your project to the latest version. This bug is probably fixed since a long time :)
See you
Romain

**Comment by Benjamin8585** [comment]
Thanks for the reply.

My title was confusing but I am talking about the current version (6.126.1).
I just mentionned the version when this function has been changed and broke my custom ContextPipe

**Reply by Romakita** [reply]
Ok. The in parameter is for swagger. I need to have your custom pipe implementation to understand why you have an issue. A repo example would be great.

see you

**Reply by Benjamin8585** [reply]
Here are my custom Context and ContextFilter. I can't provide all details of imports about User as it would be fastidious (I've created a layer on top of your platform and User is generic)

```typescript
export class OAContext<User extends BaseUser = BaseUser> extends Context {

  constructor(user?: User, lang?: string) {
    super(user, lang);
  }

  isUser(): boolean {
    return this.user?.collection.name == 'user';
  }
}

export class UserContext extends OAContext {

}
```

```typescript
@Injectable()
export class OAContextParamsPipe extends ContextPipe {

  transform(value: AnyToPromiseWithCtx, metadata: ParamMetadata) {
    const locals = value.$ctx.data;
    const context = super.transform(value, metadata);

    if (locals && locals.token && locals.token.type) {
      if (locals.token.type == 'user') {
        return new OAContext<User>(locals.user, context.lang);
      }
      // } else if (locals.token.type == 'application') {
      //   return new OAContext<User>(locals.application, context.lang);
      // }
    }
    return new OAContext<BaseUser>(context.user, context.lang);
  }
}

export function OAContextParams(): Function {
  return UsePipe(OAContextParamsPipe);
}
```

Hope you will find something I'm doing wrong. 

Thanks for the amazing work you are putting in TsED

**Reply by Romakita** [reply]
Hum ok I need also an example on his usage in the controller. Because isn't clear why you have this error. It think the problem is related to a combination on pipe and decorator :)

**Reply by Benjamin8585** [reply]
Here is a sample usage:
```typescript
@Controller('/applications')
export class ApplicationController {

  @Inject() service: ApplicationService;

  @Get('/')
  @(Returns(200, Array).Of(ApplicationSchema))
  readByUser(@OAContextParams() context: OAContext) {
    return this.service.getByUserId(context.user._id);
  }
}
```
You're probably right. I was using decorator but had to update with Pipe on v6 in order to make it work (probably I am doing it wrong?)

Thanks again

**Reply by Romakita** [reply]
Hello,
Regarding your latest example. I'm not really able to see where is the problem.

If you have a time to create a small repository to reproduce your bug (`tsed init .`), I'll take a time too inspect your issue.

See you
Romain
